contrib/caldav/calcurse-caldav.py

```diff
@@ -242,6 +328,10 @@ def push_object(conn, objhash):
             etag = next(iter(etagdict.values()))
     etag = etag.strip('"')
 
+    if authmethod == 'oauth2':
+        # Return href to a format recognized by the syncdb by removing hostname and encoding
+        href = urllib.parse.quote(uri_end + objhash + ".ics")
+
```

Comment:
Again, could you please explain why this is needed? I would not have expected any special code in the synchronization routines...

Reply:
I am not sure if this is an issue specific to the httplib2/Google Calendar, but when pushing objects, the URI path is saved to the syncdb without URL encoding. The next time syncing, it will delete everything pushed locally and pull it from the server, because it sees the encoded version from the server and the unencoded version locally as two different objects. I prevented this from happening by adding in this bit. At the very least I should change the comment. It is not very helpful in understanding why that line is there.

Reply:
Also sounds like something that should be investigated and fixed properly (in a separate patch, since it is completely unrelated to OAuth2) :)